TACC-Cloud/geoapi
# task/WG-295-WG-179-WG-185: fix and improve projects with duplicate system path constraints

## Overview: ##

This PR does the following for users:
* allows users to create multiple maps for the same system/path as long as watch_content is False.
* ensure that the user is properly notified (along with https://github.com/TACC-Cloud/hazmapper/pull/255) that they can't create a map syncing the same folder.
* users are now synced every 30 minutes (file syncing still once an hour).

This PR also refactors the code by:
* Dropping the use of ObservableProject and adding a watch_users and watch_content to Project [WG-185](https://tacc-main.atlassian.net/browse/WG-185)
* Improves nginx conf so we see the 409 error in local development.

Related to https://github.com/TACC-Cloud/hazmapper/pull/255


## Related Jira tickets: ##

* [WG-295](https://tacc-main.atlassian.net/browse/WG-295)
* [WG-179](https://tacc-main.atlassian.net/browse/WG-179)
* [WG-185](https://tacc-main.atlassian.net/browse/WG-185)


## Summary of Changes: ##

## Testing Steps: ##
1.  Test with https://github.com/TACC-Cloud/hazmapper/pull/255
2. Try creating multiple maps where `Sync Folder` is not checked.  Having two maps in the same directory should now be allowed
3. Try creating two maps for the same folder and have `Sync Folder` checked.  The creation of the second map should fail.

## Notes: ##

TODO
- [x] separate syncing users and syncing files
- [x] add script to confirm that migration works and everything looks right.
- [x] try locally with production ~and staging~ data to ensure migration is functioning.
